ts/store/reducers/wallet/cards.ts

```diff
@@ -77,7 +77,7 @@ const reducer = (
     return {
       ...state,
       selectedCardId: some(
-        typeof action.payload === "number" ? action.payload : action.payload.id
+        CreditCardId.is(action.payload) ? action.payload : action.payload.id
```

Comment:
exactly... perhaps we could use two different actions, when is one version used instead of the other?